examples/chat/basic/openai/app.py

```diff
@@ -0,0 +1,36 @@
+# pyright: basic
+from openai import AsyncOpenAI
+
+from shiny.express import ui
+
+ui.page_opts(
+    title="Hello OpenAI Chat",
+    fillable=True,
+    fillable_mobile=True,
+)
+
+# Create a chat instance, with an initial message
+chat = ui.Chat(
+    id="chat",
+    messages=[
+        {"content": "Hello! How can I help you today?", "role": "assistant"},
+    ],
+    # assistant_response_transformer=lambda x: HTML(f"<h1>{x}</h1"),
+)
+
+# Display the chat
+chat.ui()
+
+# Create the LLM client (assumes OPENAI_API_KEY is set in the environment)
```

Comment:
It would help to have API key setup instructions at the top of each app -- without that information up front, there's a lot of friction in trying to just run each app.

And in addition, it would also be good if there was a README in the examples/chat directory that explained how to set these environment variables and how to get the keys from the providers.

Reply:
It would be great if the set up instructions could also show how to use the azure end points

Reply:
@iainmwallace I assume you're referring to [these Azure docs](https://learn.microsoft.com/en-us/azure/ai-services/openai/quickstart?tabs=command-line%2Cpython-new&pivots=programming-language-python#create-a-new-python-application)? If you start from `./examples/chat/basic/openai/app.py`, it's hopefully pretty straightforward to change `AsyncOpenAI()` to `AzureOpenAI()`?

Reply:
One more idea: use the `python-dotenv` package, similar to how it's done in https://github.com/jcheng5/multimodal/blob/main/query.py.

This would allow users to create a `.env` file with the key, and then the app would automatically use that file if present; if they don't want to create a `.env` file, then setting the environment variable will still work.

For this it would be good to also add `.env` to a `.gitignore` in the examples/chat/ directory.